presto-orc/src/main/java/com/facebook/presto/orc/stream/LongInputStream.java

```diff
@@ -19,36 +19,88 @@
 
 import java.io.IOException;
 
+import static com.google.common.base.Preconditions.checkPositionIndex;
+import static java.lang.Math.toIntExact;
+
 public interface LongInputStream
         extends ValueInputStream<LongStreamCheckpoint>
 {
     long next()
             throws IOException;
 
-    void nextIntVector(int items, int[] vector)
-            throws IOException;
+    default void nextIntVector(int items, int[] vector, int offset)
```

Comment:
I think an interface is generally better.  We can always switch it to an abstract class later if we change our minds.

Reply:
It can be both. I think it's reasonable to leave it as is (an interface).

Java standard library introduces convenient methods (like `compute*` in `Map`) as default methods in interface. These `default` methods here are more or less convenient methods. So it makes sense.